host/class/uac/usb_host_uac/include/usb/uac.h

```diff
@@ -0,0 +1,480 @@
+    UAC_TYPE_I_PCM                                    = 0x0001,
+    UAC_TYPE_I_PCM8                                   = 0x0002,
+    UAC_TYPE_I_IEEE_FLOAT                             = 0x0003,
+    UAC_TYPE_I_ALAW                                   = 0x0004,
+    UAC_TYPE_I_MULAW                                  = 0x0005
+} uac_type_i_format_t;
+
+/**
+ * @brief Format Type Codes
+ *
+ */
+typedef enum {
+    UAC_FORMAT_TYPE_UNDEFINED                         = 0x0000,
+    UAC_FORMAT_TYPE_I                                 = 0x0001,
+    UAC_FORMAT_TYPE_II                                = 0x0002,
+    UAC_FORMAT_TYPE_III                               = 0x0003
+} uac_format_type_t;
+
+/**
+ * @brief Audio Class-Specific Interface Descriptor
+ *
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+} __attribute__((packed)) uac_desc_header_t;
+
+/**
+ * @brief Audio Class-Specific AC Interface Header Descriptor (bInCollection=2)
+ *
+ * @see Table 4-2 of audio10.pdf
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+    uint16_t bcdADC;
+    uint16_t wTotalLength;
+    uint8_t bInCollection;
+    uint8_t baInterfaceNr[2];
+} __attribute__((packed)) uac_ac_header_desc_t;
+
+/**
+ * @brief Audio Class-Specific AC Input Terminal Descriptor
+ *
+ * @see Table 4-3 of audio10.pdf
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+    uint8_t bTerminalID;
+    uint16_t wTerminalType;
+    uint8_t bAssocTerminal;
+    uint8_t bNrChannels;
+    uint16_t wChannelConfig;
+    uint8_t iChannelNames;
+    uint8_t iTerminal;
+} __attribute__((packed)) uac_ac_input_terminal_desc_t;
+
+/**
+ * @brief Audio Class-Specific AC Output Terminal Descriptor
+ *
+ * @see Table 4-4 of audio10.pdf
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+    uint8_t bTerminalID;
+    uint16_t wTerminalType;
+    uint8_t bAssocTerminal;
+    uint8_t bSourceID;
+    uint8_t iTerminal;
+} __attribute__((packed)) uac_ac_output_terminal_desc_t;
+
+/**
+ * @brief Audio Class-Specific AC Feature Unit Descriptor (ch=2, bControlSize=2)
+ *
+ * @see Table 4-7 of audio10.pdf
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+    uint8_t bUnitID;
+    uint8_t bSourceID;
+    uint8_t bControlSize;
+    uint8_t bmaControls[3 * 2];
+    uint8_t iFeature;
+} __attribute__((packed)) uac_ac_feature_unit_desc_t;
+
+/**
+ * @brief Audio Class-Specific AS General Descriptor
+ *
+ * @see Table 4-19 of audio10.pdf
+ */
+typedef struct {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bDescriptorSubtype;
+    uint8_t bTerminalLink;
+    uint8_t bDelay;
+    uint16_t wFormatTag;
+} __attribute__((packed)) uac_as_general_desc_t;
+
+#define UAC_FREQ_NUM_MAX                  3
```

Comment:
Can you please add a comment here? Explaining why `UAC_FREQ_NUM_MAX` equals to 3.

Reply:
There is no limitations on the number of `samples frequence` uac stream interface can offer. But for UAC1.0 devices, the number is usually not more than three. The best solution is dynamically malloc space for the supported frequencies, but this complicates the API, requiring the user to read the length first and then read the exact value.

Any suggestions? how about make it configurable in Kconfig?

Reply:
Now user can config the value with Kconfig